code/modules/projectiles/guns/beam_rifle.dm

```diff
@@ -37,13 +40,14 @@
 	var/aiming_time_left = 7
 	var/aiming_time_increase_user_movement = 3
 	var/scoped_slow = 1
-	var/aiming_time_increase_angle_multiplier = 0.6
+	var/aiming_time_increase_angle_multiplier = 0.3
 
 	var/lastangle = 0
+	var/aiming_lastangle = 0
 	var/mob/current_user = null
-	var/list/obj/effect/temp_visual/current_tracers = list()
+	var/obj/effect/projectile_beam/current_tracer
 
-	var/structure_piercing = 2				//This doesn't always work!
+	var/structure_piercing = 4				//Amount * 2. For some reason structures aren't respecting this unless you have it doubled. Probably with the objects in question's Bump() code instead of this but I'll deal with this later.
```

Comment:
it might be bumping against the turf, or itself, or perhaps it bumps on enter and then on exit